### PR TITLE
Browser fields enhancements

### PR DESCRIPTION
**Pull request by elpaso**
- Check field `subType` for a fallback icon for fields where type is not known (for instance "name"  for PG 12 table/schema identifiers
- Fix unreported issue with `fields()` not retrieved for certain tables due to multiple geometry column types/srid

### CODE REVIEW

File: tests/src/python/test_qgsproviderconnection_postgres.py [hunk 1]
```diff
@@ -329,6 +329,30 @@ def test_fields(self):
         fields = conn.fields('qgis_test', 'someData')
         self.assertEqual(fields.names(), ['pk', 'cnt', 'name', 'name2', 'num_char', 'dt', 'date', 'time', 'geom'])
 
+   def test_fields_no_pk(self):
```
Comment (elpaso) [inline tests/src/python/test_qgsproviderconnection_postgres.py:332]:
```suggestion
    def test_fields_no_pk(self):
```

File: src/providers/postgres/qgspostgresproviderconnection.cpp [hunk 2]
```diff
@@ -656,3 +656,48 @@ QList<QgsVectorDataProvider::NativeType> QgsPostgresProviderConnection::nativeTy
...
+        TableProperty::GeometryColumnType geomCol { tableInfo.geometryColumnTypes().first() };
+        tUri.setGeometryColumn( tableInfo.geometryColumn() );
+        tUri.setWkbType( geomCol.wkbType );
+        tUri.setSrid( QString::number( geomCol.crs.postgisSrid() ) );
+        if ( tableInfo.primaryKeyColumns().count() > 0 )
+        {
+          tUri.setKeyColumn( tableInfo.primaryKeyColumns().first() );
+        }
+        tUri.setParam( QStringLiteral( "checkPrimaryKeyUnicity" ), QLatin1String( "0" ) );
+        QgsVectorLayer::LayerOptions options { true, true };
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresproviderconnection.cpp:688]:
```suggestion
        QgsVectorLayer::LayerOptions options { false, false };
```

Comment (elpaso) [reply src/providers/postgres/qgspostgresproviderconnection.cpp:688]: Why `readExtentFromXml`? I'm only interested in the fields here, we don't need any extent or style information, I thought that by constructing the options with `readExtentFromXml`=`false` we can speed things up, you are right about `loadDefaultStyle` though, I think it skips some loading operations.

Comment (nyalldawson) [reply src/providers/postgres/qgspostgresproviderconnection.cpp:688]: Ok good point -- readExtentFromXml SHOULD be true!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**